apps/next/src/pages/Onboarding/flows/CreateNewWallet/CreateNewWalletFlow.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect } from 'react';
+import { WalletType } from '@avalabs/types';
+import { Route, Switch, useHistory } from 'react-router-dom';
+
+import { useOnboardingContext } from '@core/ui';
+
+import {
+  ProvideWalletDetailsScreen,
+  SelectAvatarScreen,
+  EnjoyYourWalletScreen,
+} from '../../common-screens';
+import { OnboardingModal } from '@/components/OnboardingModal';
+import { NewSeedphraseScreen } from './screens/NewSeedphraseScreen';
+import { ConfirmSeedphraseScreen } from './screens/ConfirmSeedphraseScreen';
+
+const BASE_PATH = '/onboarding/create';
+
+export const CreateNewWalletFlow = () => {
+  const history = useHistory();
+  const { setOnboardingWalletType, mnemonic, onboardingState } =
+    useOnboardingContext();
+
+  useEffect(() => {
+    setOnboardingWalletType(WalletType.Mnemonic);
+  }, [setOnboardingWalletType]);
+
+  useEffect(() => {
+    // If at any of the screens below the user does not have the required state,
+    // restart the onboarding flow.
+    if (!mnemonic && !onboardingState.isOnBoarded) {
+      history.replace(BASE_PATH);
+    }
+  }, [mnemonic, history, onboardingState.isOnBoarded]);
+
+  return (
+    <OnboardingModal
+      open
+      withCoreLogo
+      withAppInfo
+      withLanguageSelector
+      onBack={history.goBack}
+      onClose={() => history.push('/')}
+    >
+      <Switch>
+        <Route exact path={BASE_PATH}>
+          <NewSeedphraseScreen
+            step={1}
+            totalSteps={5}
```

Comment:
might worth to extract this into a variable since it should be the same everywhere and we are using it multiple time 🤔

Reply:
Good idea 👍 ea0d12a16